### DISCUSSION TITLE
import meshplot and show it in the QtWidget

### DISCUSSION THREAD

**Discussion by GavinNEU** [pyvista/pyvista#3240] [Q&A] [answered]
Dear everyone,

I would like to show an imported meshplot by using the QtInteractor. However, the plot generates a new form rather than being plugged in the QtWidget. Any suggestion has been highly appreciated in advance.

![question](https://user-images.githubusercontent.com/112489786/187447671-26c36d0a-3fa1-45ca-9706-5cfd2a4885f4.jpg)

import sys

# Setting the Qt bindings for QtPy
import os
os.environ["QT_API"] = "pyqt5"
from qtpy import QtWidgets

from pyvistaqt import QtInteractor, MainWindow
class MyMainWindow(MainWindow):

    def __init__(self, parent=None, show=True):
        QtWidgets.QMainWindow.__init__(self, parent)

        # create the frame
        self.frame = QtWidgets.QFrame()
        vlayout = QtWidgets.QVBoxLayout()

        # add the pyvista interactor object
        self.plotter = QtInteractor(self.frame)
        vlayout.addWidget(self.plotter.interactor)
        # self.signal_close.connect(self.plotter.close)

        self.frame.setLayout(vlayout)
        self.setCentralWidget(self.frame)

        # simple menu to demo functions
        mainMenu = self.menuBar()
        meshMenu = mainMenu.addMenu('Mesh')

        # allow adding an ant
        self.add_ant_action = QtWidgets.QAction('Add Ant', self)
        self.add_ant_action.triggered.connect(self.add_ant)
        meshMenu.addAction(self.add_ant_action)

        if show:
            self.show()

    def add_ant(self):
        from pyvista import examples
        mesh = examples.load_ant()
        self.plotter.add_mesh(mesh.plot())
        
if __name__ == '__main__':
    app = QtWidgets.QApplication(sys.argv)
    window = MyMainWindow()
    sys.exit(app.exec_())

**Comment by banesullivan** [comment, marked as answer]
Perhaps change `self.plotter.add_mesh(mesh.plot())` to `self.plotter.add_mesh(mesh)`. That `mesh.plot()` call is what's causing a second window to pop up

**Comment by GavinNEU** [comment]
Supper great! Problem solved! Thank you very much!!
